edi_storage_oca/models/edi_exchange_type.py

```diff
@@ -19,3 +25,44 @@ class EDIExchangeType(models.Model):
         "the files to be fetched from the pending directory in the related "
         "storage. E.g: `.*my-type-[0-9]*.\\.csv`"
     )
+
+    @api.depends("advanced_settings")
+    def _compute_storage_settings(self):
+        for rec in self:
+            rec.storage_settings = rec.advanced_settings.get("storage", {})
+
+    def _get_exchange_type_path(self):
+        """Retrieve specific path for current exchange type.
+
+        In your exchange type you can pass this config:
+
+            storage:
+                # simple string
+                path: path/to/file
+
+        Or
+
+            storage:
+                # name of the param containing the path
+                path_config_param: path/to/file
+
+        Thanks to the param you could even configure it by env.
+        """
+        self.ensure_one()
+        path = self.storage_settings.get("path")
+        if path:
+            return PurePath(path)
+        path_config_param = self.storage_settings.get("path_config_param")
+        if path_config_param:
+            icp = self.env["ir.config_parameter"].sudo()
+            path = icp.get_param(path_config_param)
+            if path:
+                return PurePath(path)
+
+    def _get_full_exchange_type_path(self, directory=None):
+        self.ensure_one()
+        path_prefix = self._get_exchange_type_path()
+        path = PurePath((directory or "").strip().rstrip("/"))
+        if path_prefix:
```

Comment:
isn't this the same done in `_make_remote_file_path`?

Reply:
Just removed `_make_remote_file_path` to avoid duplicates parts of code.